src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -148,6 +165,34 @@ private <T> List<T> loadData(final String configKey, Type type)
 		}
 	}
 
+	private List<InventorySetup> loadV2Setups() {
+		Type setupTypeV2 = new TypeToken<ArrayList<InventorySetupSerializable>>()
+		{
+
+		}.getType();
+		List<InventorySetupSerializable> issList = new ArrayList<>(loadData(CONFIG_KEY_SETUPS_V2, setupTypeV2));
+		List<InventorySetup> isList = new ArrayList<>();
+		for (final InventorySetupSerializable iss : issList)
+		{
+			isList.add(InventorySetupSerializable.convertToInventorySetup(iss));
+		}
+		return isList;
+	}
+
+	private List<InventorySetup> loadV3Setups() {
+		final String wholePrefix = ConfigManager.getWholeKey(CONFIG_GROUP, null, CONFIG_KEY_SETUPS_V3_PREFIX);
+		final List<String> loadedSetupWholeKeys = configManager.getConfigurationKeys(wholePrefix);
+		Set<String> loadedSetupKeys = loadedSetupWholeKeys.stream().map(key -> key.substring(wholePrefix.length()-CONFIG_KEY_SETUPS_V3_PREFIX.length())).collect(Collectors.toSet());
+
+		List<InventorySetup> loadedSetups = new ArrayList<>();
+		for (final String configKey : loadedSetupKeys)
+		{
+			final String storedData = configManager.getConfiguration(CONFIG_GROUP, configKey);
+			loadedSetups.add(InventorySetupSerializable.convertToInventorySetup(gson.fromJson(storedData, InventorySetupSerializable.class)));

```

Comment:
So far the only thing missing here that was different in V2 was some sort of error handling. If this line fails, the whole plugin will fail. I'm not sure if we actually need error handling in a practical sense, but at least if there is some bug or data corruption or something like that, we can get a log message when the `gson.fromJson` call fails

Long story short: I think some try catch should be added here in case the gson fails

Reply:
The old handling worried me because on failure it basically wiped everything. Should the new one be the same way?

Reply:
hmm yeah I see your point. Well in this case we can only wipe one inventory setup and try to continue. Or we let the plugin crash? I'm not really keen on that one, but at least if it does crash it would cause people to make an issue pretty quickly lol. Maybe do a try catch and give a better error message if the gson does fail then rethrow it?

Reply:
I'd honestly rather the plugin halt execution, (as it would if gson threw an exception,) because the next time data is saved to disk, every setup is wiped.

Reply:
Also I would (perhaps naively) assume that an exception from gson would show in runelite logs even without us doing something special; because it would cause the whole plugin to fail. Only better thing I can think of would be to halt execution and have the plugin panel tell the user their data is corrupt, and give them a button to export/wipe it. IMO sounds like a new PR though.

Reply:
I wrote those last two comments without seeing your reply; forgot I had to refresh to get GitHub to update.

I think that catch, log, rethrow would be good for now. That there aren't a bunch of "all my setups have been wiped" issues means gson errors are rare. If people start reporting that they encounter the error; then creating a pop up or panel could make sense.

Reply:
done; added `log.error(String.format("Exception occurred while loading %s", configKey), e);` before rethrowing.